### ISSUE TITLE
It isnt working

### ISSUE THREAD

**Issue by Juliaria08** [MaKaNu/APS#1]
```
PORT SCANNED: 22 :: with Service: ssh
Local-Adress           Foreign-Adress         Status  PID      Process
error: process ID list syntax error

Usage:
 ps [options]

 Try 'ps --help <simple|list|output|threads|misc|all>'
  or 'ps --help <s|l|o|t|m|a>'
 for additional help text.

For more details see ps(1).
0.0.0.0:22             0.0.0.0:*              28422/sshd [list
error: process ID list syntax error

Usage:
 ps [options]

 Try 'ps --help <simple|list|output|threads|misc|all>'
  or 'ps --help <s|l|o|t|m|a>'
 for additional help text.

For more details see ps(1).
```

**Comment by MaKaNu** [comment]
Could you checkout branch ```fix_#1```

I forgot to read the pid variable in quotes, which worked in my bash.

Could you additional check that your sshd still runs under PID 28422 with top/htop?
if so, what is the output of ```ps -p 28422 -o command```?

**Comment by Juliaria08** [comment]
```
git clone gh:MaKaNu/APS
git checkout fix_#1
chmod +x aps
./aps
ps -p newpid_because_reboot -o sshd
```
It still says the error:
```
APS: error: list of process IDs must follow -p
PS:  error: unknown user-defined format specifier "sshd"
```
## Running Distro/OS
`Linux art 5.11.14-artix1-1 #1 SMP PREEMPT Wed, 14 Apr 2021 14:09:15 +0000 x86_64 GNU/Linux`
### Shell
`zsh`, the shell may be the issue but shouldn't because im running as root via `sudo` and it should use root's default shell

**Comment by MaKaNu** [comment]
could you provide the output of:
```bash --version```
and what shell is bound to /bin/bash ?
my user shell is also zsh at the moment, but the script schould be called in the correct shell because of the shebang.

Could you also retry the ```ps``` command (you have a small typo):
```ps -p <your pid> -o command```
try exactly this and just change the ```<your pid>```

**Comment by Juliaria08** [comment]
```markdown
GNU bash, versión 5.1.4(1)-release (x86_64-pc-linux-gnu)
Copyright (C) 2020 Free Software Foundation, Inc.
Licencia GPLv3+: GPL de GNU versión 3 o posterior <http://gnu.org/licenses/gpl.html>

Esto es software libre, es libre para modificar y redistribuirlo.
No hay NINGUNA GARANTÍA, a la extensión permitida por la ley.
```
And `ps -p $mypid -o prog` should be `ps -p 28596 -o args -h`

Yes literaly `-o args` or `-o command` but i think it is tring to run it with 2+ `-p`

```
0.0.0.0:22             0.0.0.0:*              28596/sshd [list    ps -p [list -o command
```
Oh it is tring to do -p [list...

**Comment by MaKaNu** [comment]
Okay from that point I am shure it is not a problem with zsh, since the script is running -- either from the command ```sudo bash aps``` or the shebang  -- not in zsh and instead is executed with bash.

I have to agree that, I might have some problems understanding the following from you:
 
> And `ps -p $mypid -o prog` should be `ps -p 28596 -o args -h`
> 
> Yes literaly `-o args` or `-o command` but i think it is tring to run it with 2+ `-p`
> 
> ```
> 0.0.0.0:22             0.0.0.0:*              28596/sshd [list    ps -p [list -o command
> ```
> 
> Oh it is tring to do -p [list...

Just to explain how I implemented it: I use ```netstat -tulpnae | grep :22``` with following result on my system:

```bash
tcp    0    0 0.0.0.0:22    0.0.0.0:*    LISTEN    0   48697    1640/sshd: /usr/sbi
```
with ```awk``` I identifiy the last column which includes the pid in my case 1640.

The return value of ```ps -p 1640 -o command``` or ```ps -p 1640 -o args``` is the same:

```bash
COMMAND
sshd: /usr/sbin/sshd -D [listener] 0 of 10-100 startups
```
So in conclusion/compare to your last message, the extraction of the pid fails for some reason.
Could you provide the results of the following command and leave out every personal informations (the broadcasting should row should be enough):

```bash
sudo netstat -tulpnae | grep ":22" | awk '/tcp/{print $9}' | cut -d'/' -f1 
sudo netstat -tulpnae | grep ":22" | awk '/tcp/{print $9}'
sudo netstat -tulpnae | grep ":22"
```

**Comment by Juliaria08** [comment]
> Okay from that point I am shure it is not a problem with zsh, since the script is running -- either from the command `sudo bash aps` or the shebang -- not in zsh and instead is executed with bash.
> 
> I have to agree that, I might have some problems understanding the following from you:
> 
> > And `ps -p $mypid -o prog` should be `ps -p 28596 -o args -h`
> > Yes literaly `-o args` or `-o command` but i think it is tring to run it with 2+ `-p`
> > ```
> > 0.0.0.0:22             0.0.0.0:*              28596/sshd [list    ps -p [list -o command
> > ```
> > 
> > 
> > Oh it is tring to do -p [list...
> 
> Just to explain how I implemented it: I use `netstat -tulpnae | grep :22` with following result on my system:
> 
> ```shell
> tcp    0    0 0.0.0.0:22    0.0.0.0:*    LISTEN    0   48697    1640/sshd: /usr/sbi
> ```
> 
> with `awk` I identifiy the last column which includes the pid in my case 1640.
> 
> The return value of `ps -p 1640 -o command` or `ps -p 1640 -o args` is the same:
> 
> ```shell
> COMMAND
> sshd: /usr/sbin/sshd -D [listener] 0 of 10-100 startups
> ```
> 
> So in conclusion/compare to your last message, the extraction of the pid fails for some reason.
> Could you provide the results of the following command and leave out every personal informations (the broadcasting should row should be enough):
> 
> ```shell
> sudo netstat -tulpnae | grep ":22" | awk '/tcp/{print $9}' | cut -d'/' -f1 
> sudo netstat -tulpnae | grep ":22" | awk '/tcp/{print $9}'
> sudo netstat -tulpnae | grep ":22"
> ```


```bash
uesr@machine $ sudo netstat -tunpnae | grep :22
tcp        0      0 0.0.0.0:22              0.0.0.0:*               LISTEN      28398/sshd -D [list
tcp        0      0 10.10.10.10:50962       10.9.0.2:22             ESTABLISHED 19522/ssh
tcp        0      0 10.10.10.10:34984       213.251.45.23:22        ESTABLISHED 1926/ssh
tcp        0      0 :::22                   :::*                    LISTEN      28398/sshd -D [list
```

```bash
user@machine $ printf "[-] First
";sudo netstat -tulpnae | grep ":22" | awk /tcp/{print } | cut -d/ -f1;printf "[-] Second
";sudo netstat -tulpnae | grep ":22" | awk /tcp/{print };printf "[-] Third
";sudo netstat -tulpnae | grep ":22"
[-] First
[list


[list
[-] Second
[list


[list
[-] Third
tcp        0      0 0.0.0.0:22              0.0.0.0:*               LISTEN      28398/sshd -D [list
tcp        0      0 10.10.10.10:50962       EXTERNAL_IP:22             ESTABLISHED 19522/ssh
tcp        0      0 10.10.10.10:34984       EXTERNAL_IP:22        ESTABLISHED 1926/ssh
tcp        0      0 :::22                   :::*                    LISTEN      28398/sshd -D [list
```

For some reason `sudo netstat -tulpnae | grep ":22" | awk '/tcp/{print $7}' | cut -d'/' -f1` works

**Comment by MaKaNu** [comment]
Okay I think I found the problem. For some reasons, which do not understand completly your netstat output is missing the columns "User" and "Inode". I have an Idea to fix this.

**Comment by MaKaNu** [comment]
I think I found a fix for the problem. I reduced the netstat args to ```netstat -tpna``` which results in a similar output of your netstat call and I removed the grep and included the port grep into the awk regex.

Could you give it another try on the fix_#1 branch?

**Comment by Juliaria08** [comment]
Working, it is supposed to show established connections?

> I think I found a fix for the problem. I reduced the netstat args to `netstat -tpna` which results in a similar output of your netstat call and I removed the grep and included the port grep into the awk regex.
> 
> Could you give it another try on the fix_#1 branch?

Is it supposed to show a remote ip of a running ssh? If so working

**Comment by MaKaNu** [comment]
Yes it is supposed. Nice to hear that it fixed it. Will merge into main.

**Comment by Juliaria08** [comment]
> Yes it is supposed. Nice to hear that it fixed it. Will merge into main.

Thanks for the script, is so usefull

**Comment by MaKaNu** [comment]
Thx You are welcome. Thanks for the contribution.

**Comment by Juliaria08** [comment]
NP ( No Problem )
